cpp/leaky_core_error.h

```diff
@@ -16,61 +16,27 @@ using syntax_asdl::Token;
 using syntax_asdl::word_part_t;
 using syntax_asdl::word_t;
 
-class Usage : public std::exception {
+class Usage : public std::exception, Obj {
```

Comment:
We might be able to avoid multiple inheritance by using the `OBJ_HEADER()` macro

I'm honestly not sure if multiple inheritance gives the desired layout -- it could be worth writing a test for this